File: composeApp/src/commonMain/kotlin/com/kakapo/sudokukmp/component/control/GameKeyboardView.kt
package com.kakapo.sudokukmp.component.control

import androidx.compose.foundation.layout.Row
import androidx.compose.material3.Surface
import androidx.compose.runtime.Composable
import com.kakapo.sudokukmp.presentation.game.GameUiState
import org.jetbrains.compose.ui.tooling.preview.Preview

@Composable
internal fun GameKeyboardView(uiState: GameUiState) {
    Row {
        uiState.numbers.forEach { number ->
            KeyboardItemView(
                number = number,
                remainingUse = uiState.remainingUse?.get(number - 1)
            )
        }
    }
}


@Composable
@Preview
internal fun GameKeyboardViewPreview() {
    Surface {
        GameKeyboardView(uiState = GameUiState())
    }
}

File: iosApp/iosApp/screen/board/component/controller/GameKeyboardView.swift
import SwiftUI

struct GameKeyboardView: View {
    
    let numbers: [Int] = [1, 2, 3, 4, 5, 6, 7, 8, 9]
    
    var body: some View {
        HStack {
            ForEach(numbers, id: \.self) { number in
                KeyboardItemView(number: number, remainingUse: nil)
            }
        }
    }
}

#Preview {
    GameKeyboardView()
}
